cdap-data-fabric/src/test/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfoTest.java

```diff
@@ -590,6 +591,35 @@ public void testMultiPathFieldLineage() {
     Assert.assertEquals(expectedOperations, outgoingOperations);
   }
 
+  @Test
+  public void testNonCycle() {
```

Comment:
can you add a test for if the generated field comes from the sink? More specifically, something like:

read a, b -> write a, b, c

This can happen if the sink always adds additional data. For example, something can write an additional 'updated_time' field or something like that.

Reply:
This kind of operation cannot be defined like this, if you like at the contract of WriteOperation: https://github.com/cdapio/cdap/blob/develop/cdap-api/src/main/java/io/cdap/cdap/api/lineage/field/WriteOperation.java#L40, the write operation can only take a list of input fields, so in order to generate c. The above operation will must look something like:
read [a,b],
transform [] -> [c],
write [read.a, read.b, transform.c]

This kind of scenario is already added in this unit test.

Reply:
To be more clear, if a sink adds a column, then that sink will need to emit that generate transform, and then add the write operation.

Reply:
I see, 👍